src/js/component.js

```diff
@@ -359,19 +365,20 @@ vjs.Component.prototype.getChild = function(name){
  * @return {vjs.Component} The child component (created by this process if a string was used)
  * @suppress {accessControls|checkRegExp|checkTypes|checkVars|const|constantProperty|deprecated|duplicate|es5Strict|fileoverviewTags|globalThis|invalidCasts|missingProperties|nonStandardJsDocs|strictModuleDepCheck|undefinedNames|undefinedVars|unknownDefines|uselessCode|visibility}
  */
-vjs.Component.prototype.addChild = function(child, options){
-  var component, componentClass, componentName;
-
-  // If child is a string, create new component with options
+Component.prototype.addChild = function(child, options){
+  let component, componentName;
```

Comment:
Variables should definitely be defined in as small a scope as possible, i.e., inside blocks, and as close to where they are needed. Sometimes, you have function-level variables. Those you could define with either a let or a var but theoretically, it would be helpful to know whether that variable is long lived, i.e., for the full function. You could signal that by declaring the variable with a var instead of a let in that particular case. I'm not sure that it's worth it yet. Just throwing out some ideas.
If you have any thoughts on it, I'd be happy to hear them :)